### ISSUE TITLE
ConstraintCollocator Sphinx docs is missing documentation of all public attributes and methods

### ISSUE THREAD

**Issue by Peter230655** [csu-hmc/opty#238]
I think, it would be helpful, if the public attributes and methods could be shown in the sphinx documentation.
Thanks!

**Comment by moorepants** [comment]
Most all of the attributes are generated on instantiation of the class, so they will need be manually added to a list of attributes in the class'es docstring.

**Comment by moorepants** [comment]
Yes, but you may want to read the numpydoc format. You'd probably add info about attributes to the primary docstring of the class. See: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes

**Comment by Peter230655** [comment]
> Yes, but you may want to read the numpydoc format. You'd probably add info about attributes to the primary docstring of the class. See: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes

I will try my luck.

**Comment by Peter230655** [comment]
It has some 33 attributes, which I will attempt to list as per the numpydoc format.
It seems to me, that the methods without docstrings are probably not of too much use to the user of opty, so I would not list them in this attempt. 
**Is this o.k.?**

**Comment by moorepants** [comment]
Note the different in an attribute and a method. Only document the attributes that do not start with `_`, i.e. the only the public methods.

**Comment by Peter230655** [comment]
Clear.
Questions:
Under Parameters you have _equations of motion_ and you explain what it is.
There is an attribute _eom_. Should this be listed or is it redundent?

Under Paramaeters you have state_symbols which is also an attribute. Should this be listed again?

To get an attribute one has to do this: Problem_instance.collocator.atribute.
I had forgotten, Timo told me again. Should this be explained?

**Comment by moorepants** [comment]
We should consistently document all public attributes. A little redundancy doesn't hurt.

**Comment by moorepants** [comment]
>  I had forgotten, Timo told me again. Should this be explained?

You could explain that in the Problem documentation.

**Comment by Peter230655** [comment]
> > I had forgotten, Timo told me again. Should this be explained?
> 
> You could explain that in the Problem documentation.

You mean in the docstrings of the class Problem, right?

**Comment by Peter230655** [comment]
> We should consistently document all public attributes. A little redundancy doesn't hurt.

Clear, I will attempt to do so.

**Comment by moorepants** [comment]
> > > I had forgotten, Timo told me again. Should this be explained?
> > 
> > 
> > You could explain that in the Problem documentation.
> 
> You mean in the docstrings of the class Problem, right? Possible in the same PR?

Yes

**Comment by Peter230655** [comment]
I believe, this may be closed now?
